### PR TITLE
scribe: init at unstable-2022-03-12

### PR DESCRIPTION
**Pull request by erdnaxe**
###### Description of changes

Scribe is an alternative frontend for Medium, focused on privacy.

Homepage: https://scribe.rip/

It is based on the Lucky framework that provides database configuration, but scribe does not use the database.

To run the server, you need to be in a folder containing a symlink to `/nix/store/xxx-scribe-unstable-2022-03-12/public` and a `config/watch.yml` configuration file containing:
```yaml
host: 127.0.0.1
port: 5000
```

I might add a NixOS module and test to this PR later.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/scribe/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  inherit pname version src;
+
+  format = "crystal";
+  shardsFile = ./shards.nix;
+  crystalBinaries.scribe.src = "src/scribe.cr";
+  crystalBinaries.lucky_tasks.src = "tasks.cr";
+
+  buildInputs = [ openssl ];
+  doCheck = false; # require postgresql
+  doInstallCheck = false; # freeze
```
Comment (SuperSandro2000) [inline pkgs/servers/scribe/default.nix:39]: freeze?

File: pkgs/servers/scribe/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  preBuild = ''
+    # Build requires public/mix-manifest.json
+    cp -r ${publicFiles}/* public
```
Comment (SuperSandro2000) [inline pkgs/servers/scribe/default.nix:43]: Could we symlink instead?

**Comment by JohnRTitor** [comment]
Hello, could you rebase and respond to the questions please?

**Comment by SuperSandro2000** [comment]
Closing because this is pretty stale.

**Pull request closed without merging**